core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -124,6 +128,7 @@ public GenericManifestFile(String path, long length, int specId, ManifestContent
     this.deletedRowsCount = deletedRowsCount;
     this.partitions = partitions == null ? null : partitions.toArray(new PartitionFieldSummary[0]);
     this.fromProjectionPos = null;
+    this.keyMetadata = ByteBuffers.toByteArray(keyMetadata);
```

Comment:
Do we need a null guard here?

Reply:
There is already a null guard in `ByteBuffers`